src/frame_buffer.rs

```diff
@@ -403,7 +403,7 @@ unsafe impl<T: PixelData> PixelStruct for T {
 
 macro_rules! offset_of {
     ($ty:ty, $field:tt) => {
-        unsafe { &(*(0 as *const $ty)).$field as *const _ as usize }
+        unsafe { &(*(ptr::null() as *const $ty)).$field as *const _ as usize }
```

Comment:
I think this would make the macro unusable in a const context, which would interfere with a transition to employing associated consts. Is that justified?

Reply:
I'm not sure I understand what you mean here, ptr::null still gives a `*const T` so I think it'd be ok in a const context? I'm not as familiar with the macros in Rust, an example might help some. I can switch this back as well to 0.

Reply:
`ptr::null` returns a `*const T`, but is not itself a `const fn`, hence it cannot be used to initialize a constant.

Reply:
Ah I see, I'll switch this back